File: mvp-arch-component/app/src/main/java/thaihn/android/architecturecomponent/MyApplication.java
package thaihn.android.architecturecomponent;

import android.app.Application;
import thaihn.android.architecturecomponent.data.room.db.AppDatabase;

public class MyApplication extends Application {

    private static MyApplication sInstance;
    private AppDatabase mAppDatabase;

    public static MyApplication getInstance() {
        return sInstance;
    }

    public AppDatabase getAppDatabase() {
        return mAppDatabase;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        mAppDatabase = AppDatabase.getInstance(this);
        sInstance = this;
    }
}

File: clear-architect/presentation/src/main/java/thaihn/com/androidcleanarchitecture/MyApplication.kt
package thaihn.com.androidcleanarchitecture

import android.app.Application
import thaihn.com.androidcleanarchitecture.di.component.DaggerAppComponent

class MyApplication : Application() {

  override fun onCreate() {
    super.onCreate()
    DaggerAppComponent.builder()
        .application(this)
        .build()
        .inject(this)
  }
}